airflow/ui/src/queries/usePatchTaskInstanceDryRun.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useQuery, type UseQueryOptions } from "@tanstack/react-query";
+
+import { TaskInstanceService } from "openapi/requests/services.gen";
+import type { PatchTaskInstanceBody, PatchTaskInstanceDryRunResponse } from "openapi/requests/types.gen";
+
+type Props<TData, TError> = {
+  dagId: string;
+  dagRunId: string;
+  mapIndex: number;
+  options?: Omit<UseQueryOptions<TData, TError>, "queryFn" | "queryKey">;
+  requestBody: PatchTaskInstanceBody;
+  taskId: string;
+};
+
+export const usePatchTaskInstanceDryRun = <TData = PatchTaskInstanceDryRunResponse, TError = unknown>({
+  dagId,
+  dagRunId,
+  mapIndex,
+  options,
+  requestBody,
+  taskId,
+}: Props<TData, TError>) =>
+  useQuery<TData, TError>({
+    ...options,
+    queryFn: () =>
+      TaskInstanceService.patchTaskInstanceDryRun({
+        dagId,
+        dagRunId,
+        mapIndex,
+        requestBody,
+        taskId,
+      }) as TData,
+    queryKey: [
+      "patchTaskInstanceDryRun",
+      dagId,
+      dagRunId,
+      taskId,
+      mapIndex,
+      { ...requestBody, note: Boolean(requestBody.note) },
```

Comment:
Why is note a boolean?

Reply:
The idea was to cache the request on 'there is a note' or 'there is not a note' in the update request. We don't want the dry_run query to trigger on every keystroke, but I figured it's simpler to not specify the `note` at all in the caching key. It's just more verbose as I can't do anymore:
```js
{...requestBody}
```

But something like that just to omit 1 key:
```js
      requestBody.new_state,
      requestBody.include_downstream,
      requestBody.include_future,
      requestBody.include_past,
      requestBody.include_upstream,
```